### ISSUE TITLE
In macOS, cmd-x doesn't cut text, but launches counsel-M-x

### ISSUE THREAD

**Issue by tien** [doomemacs/doomemacs#7298]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27 or 28. *(Doom does not support development builds like 29+ or any version ending in .50 or .9x)*

### Expected behavior

`cmd-x` cuts the text

### Current behavior

`cmd-x` launches `counsel-M-x` (most other `cmd-` combinations are fine)

### Steps to reproduce

1. Open file
2. `cmd-x`

### System Information

https://pastebin.com/1T7ENBnY

**Comment by hlissner** [comment]
I suspect this is a local config issue, because Doom's default behavior is to [use the option key for meta keybinds](https://github.com/doomemacs/doomemacs/blob/master/lisp/doom-keybinds.el#L36-L39).

Do you have something like the following in your config?

```lisp
;; in $DOOMDIR/config.el
(setq mac-command-modifier   'meta
      ns-command-modifier    'meta)
```

I will repoen it if it turns out not to be a local config issue.

**Comment by hlissner** [comment]
Ah wait, scratch that. Forgot about [this keybind](https://github.com/doomemacs/doomemacs/blob/master/modules/config/default/config.el#L320). That said, the keybind right after it should cause it to invoke cutting if you have a selection active, is this not the case?

**Comment by jcarlos** [comment]
On my mac using latest doom emacs when a text is selected cmd-x cuts text as expected. When text is not selected cmd-x opens M-x and I quite like this behaviour as it's easier to press than alt-x.

**Comment by tien** [comment]
Yeah, you are right, cutting behave correctly when text is selected. Potentially an incorrect but possibly useful "feature not a bug" :))

**Comment by tien** [comment]
Oh actually, it doesn't cut if you are in insert mode

**Comment by filipesilva** [comment]
I always get tripped up on this too. Sometimes it cuts, sometimes it doesn't.
